src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/UpdateSentenceDateService.kt

```diff
@@ -49,9 +46,10 @@ class UpdateSentenceDateService(
 
     val sentenceDates = prisoner.sentenceDetail.toSentenceDates()
 
-    logUpdate(licenceId, licenceEntity, licenceStartDate, sentenceDates)
+    val sentenceChanges = licenceEntity.getDateChanges(sentenceDates, licenceStartDate)
+    val dateChanges = sentenceChanges.dates.filter { !it.type.hdcOnly || licenceEntity is HdcLicence }
```

Comment:
only extract date changes that are relevant for licence

Reply:
Not quite sure I understand this yet, so we are filtering out not HDC only or if it is HDC?